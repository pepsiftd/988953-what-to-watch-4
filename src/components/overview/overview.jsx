import React from 'react';
import PropTypes from 'prop-types';
import {getRatingEstimate} from '@/utils';

const Overview = ({ratingScore, ratingCount, description, director, starring}) => {
  const starringShort = starring.slice(0, 4);
  const starringString = starringShort.join(`, `);
  const score = String(ratingScore).replace(`.`, `,`);
  return (
    <React.Fragment>
      <div className="movie-rating">
        <div className="movie-rating__score">{score}</div>
        <p className="movie-rating__meta">
          <span className="movie-rating__level">{getRatingEstimate(ratingScore)}</span>
          <span className="movie-rating__count">{ratingCount} ratings</span>
        </p>
      </div>

      <div className="movie-card__text">
        {description}

        <p className="movie-card__director"><strong>{`Director: ${director}`}</strong></p>

        <p className="movie-card__starring"><strong>{`Starring: ${starringString}${starringShort.length < starring.length ? ` and other` : ``}`}</strong></p>
      </div>
    </React.Fragment>
  );
};

Overview.propTypes = {
  ratingScore: PropTypes.number.isRequired,
  ratingCount: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  director: PropTypes.string.isRequired,
  starring: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export {Overview};
