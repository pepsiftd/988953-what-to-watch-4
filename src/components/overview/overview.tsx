import * as React from 'react';
import {getRatingEstimate, getRatingString} from '@/utils';
import {MAX_ACTORS_IN_SHORT_DETAILS} from '@/const';

interface Props {
  ratingScore: number;
  ratingCount: number;
  description: string;
  director: string;
  starring: string[];
}

const Overview: React.FunctionComponent<Props> = ({ratingScore, ratingCount, description, director, starring}) => {
  const starringShort = starring.slice(0, MAX_ACTORS_IN_SHORT_DETAILS);
  const starringString = starringShort.join(`, `);
  const score = getRatingString(ratingScore);
  const estimate = getRatingEstimate(ratingScore);
  return (
    <React.Fragment>
      <div className="movie-rating">
        <div className="movie-rating__score">{score}</div>
        <p className="movie-rating__meta">
          <span className="movie-rating__level">{estimate}</span>
          <span className="movie-rating__count">{`${ratingCount} ratings`}</span>
        </p>
      </div>

      <div className="movie-card__text">
        {description}

        <p className="movie-card__director"><strong>{`Director: ${director}`}</strong></p>

        <p className="movie-card__starring"><strong>{`Starring: ${starringString}${starringShort.length < starring.length ? ` and other` : ``}`}</strong></p>
      </div>
    </React.Fragment>
  );
};

export {Overview};
