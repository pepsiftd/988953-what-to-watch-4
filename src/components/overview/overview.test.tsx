import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {Overview} from './overview';

describe(`Overview should render correctly`, () => {
  it(`with 5 or more actors`, () => {
    const tree = renderer.create(
        <Overview
          ratingScore={8.9}
          ratingCount={31337}
          description="description text"
          director="Arthur Pirozhkov"
          starring={[`Belka`, `Strelka`, `Monkey`, `Rat`, `Donkey`]}
        />
    );

    expect(tree).toMatchSnapshot();
  });

  it(`with less than 5 actors`, () => {
    const tree = renderer.create(
        <Overview
          ratingScore={8.9}
          ratingCount={31337}
          description="description text"
          director="Arthur Pirozhkov"
          starring={[`Belka`, `Strelka`, `Monkey`, `Rat`]}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
