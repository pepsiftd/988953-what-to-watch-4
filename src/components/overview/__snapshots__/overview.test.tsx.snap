// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview should render correctly with 5 or more actors 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8,9
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        31337 ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    description text
    <p
      className="movie-card__director"
    >
      <strong>
        Director: Arthur Pirozhkov
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: Belka, Strelka, Monkey, Rat and other
      </strong>
    </p>
  </div>,
]
`;

exports[`Overview should render correctly with less than 5 actors 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8,9
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        31337 ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    description text
    <p
      className="movie-card__director"
    >
      <strong>
        Director: Arthur Pirozhkov
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: Belka, Strelka, Monkey, Rat
      </strong>
    </p>
  </div>,
]
`;
